When spring.liquibase.clear-checksums property of spring boot is set, when the target database is empty, it throws exception, and SpringLiquibase.afterPropertiesSet doesn't run the changelogs

### Search first

- [X] I searched and no similar issues were found

### Description

So I want to submit two bugs, but didn't want to open two issues because two can be reproduced from the same repo that I will provide

### Steps To Reproduce

First bug: When `spring.liquibase.clear-chekcsums` is set to true, initializing on an empty database throws exception in spring boot application on application startup
Steps:
1. Clone repo: https://github.com/fatihdogmus/afterpropertiesset-bug
2. Run `docker compose up -d`. This will initialize a bitnami alpine 15.4 postgres image and bind the data to ~/DockerVolumes/liquibase-bug, since we need persistent storage for the next bug
3. Run the application
4. You will see exception: `encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: ERROR: relation "public.databasechangelog" does not exist
  Position: 8 [Failed SQL: (0) UPDATE public.databasechangelog SET MD5SUM = NULL]`

Second bug: This is a bit trickier. After DB initialization, when we want to reset the database programatically with `SpringLiquibase`, by first dropping everything, the changes are not run when `spring.liquibase.clear-chekcsums` is set to true

Steps:
1. Clone the same repo and get DB working with docker. I tried with empty database, so it would be wise to remove anything from previous DB.
2. First, set `spring.liquibase.clear-chekcsums` to false, otherwise the application doesn't start
3. Stop the application, set `spring.liquibase.clear-chekcsums` to true and run the application. After this, the DB is in correct state.
4. Send a GET request to `http://localhost:8087/resetDatabase`, which is this code:
```java
@RestController
public class ResetDatabaseController {

    private final SpringLiquibase springLiquibase;

    public ResetDatabaseController(final SpringLiquibase springLiquibase) {
        this.springLiquibase = springLiquibase;
    }

    @GetMapping("/resetDatabase")
    public String resetDatabase() throws LiquibaseException {
        springLiquibase.setClearCheckSums(false);
        springLiquibase.setDropFirst(true);

        springLiquibase.afterPropertiesSet();

        springLiquibase.setClearCheckSums(true);
        springLiquibase.setDropFirst(false);

        return "Database has been reset successfully";
    }
}

```
5. You will see Database is up to date, no changesets to execute command in the logs, after all the objects are dropped. 
 And the database won't be updated.

Strangely, this bug doesn't happen if the application is started with `spring.liquibase.clear-chekcsums` set to false. I don't know what the problem is, I set it false on my code anyway so I don't know why it would behave differently.

Both problems didn't happen before we migrated to spring boot 3.2.0 from 3.1.4

### Expected/Desired Behavior

For first bug to not throw exceptions and second bug to run the changes.

### Liquibase Version

4.24.0

### Database Vendor & Version

PostgreSQL 15.4

### Liquibase Integration

Spring Boot

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

Macos

### Additional Context

_No response_

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR (Thank you!)

I can provide a PR, but I don't know what the problem is. I've looked at the source code, but couldn't find the problem, since I'm not very familiar with the code base. If you can find the problem and direct me to it, I can submit a PR.

Hey,

Since spring boot 3.2.0 comes with 4.24.0, when I try to set version to 4.23.2 with spring boot 3.2.0, it gives the following error:
`java.lang.NoClassDefFoundError: liquibase/UpdateSummaryOutputEnum
	at java.base/java.lang.Class.getDeclaredFields0(Native Method) ~[na:na]
	at java.base/java.lang.Class.privateGetDeclaredFields(Class.java:3473) ~[na:na]
	at java.base/java.lang.Class.getDeclaredField(Class.java:2780) ~[na:na]
	at org.springframework.boot.context.properties.bind.DefaultBindConstructorProvider$Constructors.isInnerClass(DefaultBindConstructorProvider.java:144) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.context.properties.bind.DefaultBindConstructorProvider$Constructors.getCandidateConstructors(DefaultBindConstructorProvider.java:134) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.context.properties.bind.DefaultBindConstructorProvider$Constructors.getConstructors(DefaultBindConstructorProvider.java:103) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.context.properties.bind.DefaultBindConstructorProvider.getBindConstructor(DefaultBindConstructorProvider.java:55) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBean.deduceBindMethod(ConfigurationPropertiesBean.java:298) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBeanRegistrar.createBeanDefinition(ConfigurationPropertiesBeanRegistrar.java:93) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBeanRegistrar.registerBeanDefinition(ConfigurationPropertiesBeanRegistrar.java:89) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBeanRegistrar.register(ConfigurationPropertiesBeanRegistrar.java:61) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBeanRegistrar.register(ConfigurationPropertiesBeanRegistrar.java:55) ~[spring-boot-3.2.0.jar:3.2.0]
	at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[na:na]
	at org.springframework.boot.context.properties.EnableConfigurationPropertiesRegistrar.registerBeanDefinitions(EnableConfigurationPropertiesRegistrar.java:49) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.context.annotation.ImportBeanDefinitionRegistrar.registerBeanDefinitions(ImportBeanDefinitionRegistrar.java:86) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.lambda$loadBeanDefinitionsFromRegistrars$1(ConfigurationClassBeanDefinitionReader.java:376) ~[spring-context-6.1.1.jar:6.1.1]
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:986) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsFromRegistrars(ConfigurationClassBeanDefinitionReader.java:375) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsForConfigurationClass(ConfigurationClassBeanDefinitionReader.java:148) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitions(ConfigurationClassBeanDefinitionReader.java:120) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:428) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:289) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:349) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:118) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:775) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:597) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:753) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:455) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:323) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1342) ~[spring-boot-3.2.0.jar:3.2.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1331) ~[spring-boot-3.2.0.jar:3.2.0]
	at spring.liquibase.afterpropertiessetbug.AfterpropertiessetBugApplication.main(AfterpropertiessetBugApplication.java:10) ~[classes/:na]
Caused by: java.lang.ClassNotFoundException: liquibase.UpdateSummaryOutputEnum
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[na:na]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[na:na]
	... 33 common frames omitted`

it seems it is trying to find an enum that doesn't exist in older versions.

When I set the spring boot version to 3.1.6, and liquibase version to 4.23.2, the first bug goes away, but the second one, where it doesn't run the changelogs, still persists. And the workaround of setting the clear checksums to false also works.

Weirdly enough, when the application starts from an emty database and clear checksums is true, the reset database works, but when I restart the application it doesn't.

--- a/liquibase-integration-tests/src/test/groovy/liquibase/command/core/ClearCheckSumsIntegrationTest.groovy
+++ b/liquibase-integration-tests/src/test/groovy/liquibase/command/core/ClearCheckSumsIntegrationTest.groovy
@@ -18,6 +18,24 @@ class ClearCheckSumsIntegrationTest extends Specification {
     @Shared
     private DatabaseTestSystem h2 = (DatabaseTestSystem) Scope.currentScope.getSingleton(TestSystemFactory.class).getTestSystem("h2")
 
+    def "clearing checksums in an empty database does not throw an exception"() {
+        given:
+        CommandUtil.runDropAll(h2)
+
+        when:
+        def h2Database = h2.getDatabaseFromFactory()
+        def commandResults = new CommandScope("clearChecksums")
+                .addArgumentValue(DbUrlConnectionArgumentsCommandStep.URL_ARG, h2.getConnectionUrl())
+                .addArgumentValue(DbUrlConnectionArgumentsCommandStep.DATABASE_ARG, h2Database)
+                .execute()
+
+        then:
+        commandResults.results.size() == 0
+
+        cleanup:
+        CommandUtil.runDropAll(h2)
+    }
+
     def "validate checksums are cleared"() {
         given:
         def updateCommand = new CommandScope(UpdateCommandStep.COMMAND_NAME)

--- a/liquibase-standard/src/main/java/liquibase/command/core/ClearChecksumsCommandStep.java
+++ b/liquibase-standard/src/main/java/liquibase/command/core/ClearChecksumsCommandStep.java
@@ -3,7 +3,10 @@ package liquibase.command.core;
 import liquibase.Scope;
 import liquibase.changelog.ChangeLogHistoryService;
 import liquibase.changelog.ChangeLogHistoryServiceFactory;
-import liquibase.command.*;
+import liquibase.command.AbstractCommandStep;
+import liquibase.command.CommandDefinition;
+import liquibase.command.CommandResultsBuilder;
+import liquibase.command.CommandScope;
 import liquibase.database.Database;
 import liquibase.lockservice.LockService;
 
@@ -31,9 +34,10 @@ public class ClearChecksumsCommandStep extends AbstractCommandStep {
         CommandScope commandScope = resultsBuilder.getCommandScope();
         final Database database = (Database) commandScope.getDependency(Database.class);
 
-        ChangeLogHistoryService changeLogService = ChangeLogHistoryServiceFactory.getInstance().getChangeLogService(database);
+        ChangeLogHistoryService changeLogHistoryService = Scope.getCurrentScope().getSingleton(ChangeLogHistoryServiceFactory.class).getChangeLogService(database);
+        changeLogHistoryService.init();
         Scope.getCurrentScope().getLog(getClass()).info(String.format("Clearing database change log checksums for database %s", database.getShortName()));
-        changeLogService.clearAllCheckSums();
+        changeLogHistoryService.clearAllCheckSums();
     }
 
     @Override

